src/sqlfluff/api/simple.py

```diff
@@ -1,20 +1,23 @@
 """The simple public API methods."""
 
+from typing import Any, Dict, List, Optional, TextIO, Union
 from sqlfluff.core import Linter
+from sqlfluff.core import SQLBaseError
+from sqlfluff.core.linter import ParsedString
 
 
 class APIParsingError(ValueError):
     """An exception which holds a set of violations."""
 
-    def __init__(self, violations, **kwargs):
+    def __init__(self, violations: List[SQLBaseError], *args: Any):
         self.violations = violations
         self.msg = f"Found {len(violations)} issues while parsing string."
         for viol in violations:
             self.msg += f"\n{viol!s}"
-        super().__init__(self.msg, **kwargs)
+        super().__init__(self.msg, *args)
```

Comment:
Changed the kwargs to args due to this mypy error: https://github.com/python/mypy/issues/4183

Shouldn't impact current use of the exception.